### ISSUE TITLE
Profile-level skip_list is not applied for name[casing] and name[template] in basic profile

### ISSUE THREAD

**Issue by sky-joker** [ansible/ansible-lint#5087]
##### Summary

According to src/ansiblelint/data/profiles.yml, the basic profile defines the following skip_list.

https://github.com/ansible/ansible-lint/blob/main/src/ansiblelint/data/profiles.yml#L41-L43

However, when profile: basic is configured, name[casing] and name[template] are still reported as violations.
It looks like the profile-level skip_list is not applied to the runtime skip_list.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```
$ ansible-lint --version
ansible-lint 26.4.0 using ansible-core:2.21.0 ansible-compat:26.3.0 ruamel-yaml:0.19.1 ruamel-yaml-clib:0.2.15
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

```
$ cat .ansible-lint
---
profile: basic
```

Example playbook:

```yaml
---
- name: Test
  hosts: localhost
  gather_facts: false
  tasks:
    - name: "test {{ hoge }} fuga"
      ansible.builtin.debug:
        msg: OK
```

Run and Output:

```
$ ansible-lint
(snip)
name[casing]: All names should start with an uppercase letter.
playbooks/a.yml:6:13 Task/Handler: test {{ hoge }} fuga

name[template]: Jinja templates should only be at the end of 'name'
playbooks/a.yml:6:13 Task/Handler: test {{ hoge }} fuga
```

##### Desired Behavior

name[casing] and name[template] should be skipped when the basic profile is selected, because they are listed in the basic profile’s skip_list.

```
$ ansible-lint
(snip)
Passed: 0 failure(s), 0 warning(s) in 2 files processed of 2 encountered. Profile 'basic' was required, but 'production' profile passed.
```

##### Actual Behavior

name[casing] and name[template] are not skipped.
They are skipped only when they are explicitly added to the user configuration:

```
---
profile: basic

skip_list:
  - name[casing]
  - name[template]
```

or when passed with -x.

From reviewing the code, I could not find a code path that reads PROFILES[profile]["skip_list"] and merges it into the runtime skip_list.
